ingress/controllers/nginx/nginx/template.go

```diff
@@ -100,3 +107,68 @@ func toCamelCase(src string) string {
 	}
 	return string(bytes.Join(chunks, nil))
 }
+
+func buildLocation(input interface{}) string {
```

Comment:
please include a comment with an example output string so it's clear what role this function plays in producing the final template